### PR TITLE
Refacto service container

### PR DESCRIPTION
**Pull request by yohanBarilleau**
The purpose of this refactoring is to extract all the logic of the Service container patch from @Matt75  in an external service to extract it in a library

### CODE REVIEW

File: ps_checkout.php [hunk 1]
```diff
@@ -22,6 +22,7 @@
 use PrestaShop\Module\PrestashopCheckout\PayPal\Intent;
 use PrestaShop\Module\PrestashopCheckout\PayPal\Mode;
 use PrestaShop\Module\PrestashopCheckout\PayPal\PayPalConfiguration;
+use PrestaShop\ModuleLibServiceContainer\DependencyInjection\ServiceContainer;
```
Comment (Quetzacoalt91) [inline ps_checkout.php:25]: I missed this part in the last PRs, using `use [...]` won't be compliant with PrestaShop 1.6 (the class will be detected as invalid)

File: ps_checkout.php [hunk 2]
```diff
@@ -138,6 +141,8 @@ public function __construct()
 
         $this->confirmUninstall = $this->l('Are you sure you want to uninstall this module?');
         $this->ps_versions_compliancy = ['min' => '1.6.1', 'max' => _PS_VERSION_];
+
+        $this->serviceContainer = new ServiceContainer($this->name, $this->getLocalPath());
```
Comment (Quetzacoalt91) [inline ps_checkout.php:145]:
```suggestion
        $this->serviceContainer = new PrestaShop\ModuleLibServiceContainer\DependencyInjection\ServiceContainer($this->name, $this->getLocalPath());
```

File: ps_checkout.php [hunk 3]
```diff
@@ -108,12 +109,14 @@ class Ps_checkout extends PaymentModule
...
     const VERSION = '2.0.0';
 
-    /** @var \Symfony\Component\DependencyInjection\ContainerInterface */
-    private $container;
-
     /** @var \Psr\Log\LoggerInterface */
     private $logger;
 
+    /**
+     * @var ServiceContainer
```
Comment (Quetzacoalt91) [inline ps_checkout.php:116]:
```suggestion
     * @var PrestaShop\ModuleLibServiceContainer\DependencyInjection\ServiceContainer
```

File: ps_checkout.php (hunk 1 above)
Comment (Quetzacoalt91) [reply ps_checkout.php:25]:
```suggestion
```

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by Matt75 — APPROVED** [review]

**Pull request merged**